content-serve-auth.md

```diff
@@ -0,0 +1,106 @@
+# Content Server Authorization Spec
+
+## Editors
+
+- [Felipe Forbeck](https://github.com/fforbeck), [Storacha Network](https://storacha.network/)
+
+## Authors
+
+- [Felipe Forbeck](https://github.com/fforbeck), [Storacha Network](https://storacha.network/)
+
+## Abstract
+
+Content Server Authorization ensures that access to content is governed by UCAN delegations (User Controlled Authorization Network) and served by explicitly authorized services.
+This mechanism allows content owners to delegate retrieval capabilities to specific services, ensuring that only authorized entities can access the content.
+
+## Terminology
+
+- **Space**: A logical container for data, identified by a DID (Decentralized Identifier). Authorization is granted for an entire space, and all data within a space shares the same permissions.
+- **Delegation**: The act of granting specific capabilities to another entity via UCAN, and the signed document proving the delegation (also called a "**proof**").
+- **Gateway**: A service (e.g., Freeway) that facilitates content retrieval and enforces authorization policies.
+- **Delegations KV Store**: A key-value store used by the gateway to manage and validate delegations.
+
+## Delegation Flow
+
+The delegation process involves the following steps
+
+1. **Space Creation and Delegation**
+
+   - A user creates a space using the Storacha client.
+   - During space creation, the Storacha Client delegates the `space/content/serve` capability to the Storacha Gateway (e.g., Freeway) by sending a `POST /access/delegate` request containing the UCAN delegation.
+   - The Gateway validates the UCAN delegation and stores it in the Delegations KV store.
+
+2. **Delegation Storage**
+
+   - Delegations are stored using a key composed of the space DID and the delegation CID.
+   - Multiple delegations can exist for the same space/bucket, allowing flexibility in access control.
+
+### Delegation Flow Diagram
+
+```mermaid
+sequenceDiagram
+    participant User
+    participant Client
+    participant Gateway
+    participant DelegationsKV
+
+    User->>Client: Create Space (DID)
+    Client->>Gateway: POST /access/delegate (UCAN)
+    Gateway->>DelegationsKV: Store Delegation (space DID + delegation CID)
+    Gateway-->>Client: Delegation Acknowledged
+```
+
+## Content Retrieval Flow
+
+When a client requests content
+
+1. **Request Handling**
+
+   - The client sends a `GET /ipfs/:cid` request to the Gateway.
+   - The Gateway resolves the associated space/bucket and retrieves the relevant delegations from the KV store.
```

Comment:
I would avoid referencing "KV" anywhere - it doesn't have to be a key-value store, could be any database or even flat files.